src/net/sourceforge/kolmafia/swingui/panel/GearChangePanel.java

```diff
@@ -0,0 +1,1496 @@
+  private JCheckBox weapon1H;
+  private JRadioButton[] offhandTypes;
+
+  private int deferredUpdateLevel = 0;
+  private final EquipmentComboBox[] equipment;
+  private final List<SortedListModel<AdventureResult>> equipmentModels;
+
+  private final SortedListModel<FamiliarData> familiars = new SortedListModel<>();
+  private final SortedListModel<FamiliarData> crownFamiliars = new SortedListModel<>();
+  private final SortedListModel<FamiliarData> bjornFamiliars = new SortedListModel<>();
+
+  private final EquipmentPanel equipmentPanel;
+  private final CustomizablePanel customizablePanel;
+
+  private final OutfitComboBox outfitSelect, customSelect;
+  private final FamiliarComboBox familiarSelect;
+  private final ThroneComboBox crownSelect;
+  private final BjornComboBox bjornSelect;
+  private JLabel sticker1Label, sticker2Label, sticker3Label;
+  private FamLockCheckbox famLockCheckbox;
+  private FakeHandsSpinner fakeHands;
+
+  public GearChangePanel() {
+    super(new BorderLayout());
+
+    this.equipment = new EquipmentComboBox[EquipmentManager.ALL_SLOTS];
+    this.equipmentModels = new ArrayList<>(EquipmentManager.ALL_SLOTS);
+
+    List<List<AdventureResult>> lists = EquipmentManager.getEquipmentLists();
+
+    for (int i = 0; i < this.equipment.length; ++i) {
+      LockableListModel<AdventureResult> list;
+
+      // We maintain our own lists for certain slots
+      switch (i) {
+        case EquipmentManager.HAT:
+        case EquipmentManager.PANTS:
+        case EquipmentManager.SHIRT:
+        case EquipmentManager.CONTAINER:
+        case EquipmentManager.WEAPON:
+        case EquipmentManager.OFFHAND:
+        case EquipmentManager.ACCESSORY1:
+        case EquipmentManager.ACCESSORY2:
+        case EquipmentManager.ACCESSORY3:
+        case EquipmentManager.FAMILIAR:
+        case EquipmentManager.BOOTSKIN:
+        case EquipmentManager.BOOTSPUR:
+        case EquipmentManager.HOLSTER:
+          list = new SortedListModel<>();
+          this.equipmentModels.add((SortedListModel<AdventureResult>) list);
+          break;
+        default:
+          list = (LockableListModel<AdventureResult>) lists.get(i);
+          this.equipmentModels.add(null);
+          break;
+      }
+
+      this.equipment[i] = new EquipmentComboBox(list, i);
+    }
+
+    this.familiarSelect = new FamiliarComboBox(this.familiars);
+    this.crownSelect = new ThroneComboBox(this.crownFamiliars);
+    this.bjornSelect = new BjornComboBox(this.bjornFamiliars);
+    this.outfitSelect =
+        new OutfitComboBox((LockableListModel<SpecialOutfit>) EquipmentManager.getOutfits());
+    this.customSelect =
+        new OutfitComboBox((LockableListModel<SpecialOutfit>) EquipmentManager.getCustomOutfits());
+
+    this.equipmentPanel = new EquipmentPanel();
+    this.customizablePanel = new CustomizablePanel();
+
+    this.tabs = KoLmafiaGUI.getTabbedPane();
+    this.tabs.setTabPlacement(SwingConstants.BOTTOM);
```

Comment:
Please use SwingConstants.TOP, as it was before.

Reply:
Huh. I compared to my jar version of mafia to ensure it looked the same. The jar that I just downloaded has the tabs on the bottom. Maybe it's OS-dependent somehow? I don't see a preference that would make the positioning user-specific.

Reply:
Java doc for JtabbedPane.setTabPlacement()

```public void setTabPlacement(int tabPlacement)```
Sets the tab placement for this tabbedpane. Possible values are:
- JTabbedPane.TOP
- JTabbedPane.BOTTOM
- JTabbedPane.LEFT
- JTabbedPane.RIGHT

The default value, if not set, is SwingConstants.TOP.

I have no idea why it is BOTTOM for you. Perhaps it is affected by look and feel?

GenericFrame does not call that - and GearChangeFrame has tabs at the top.
In your PR, I commented out the line - and the tabs are at the top.

I think you should remove that call and just let os/look-and-feel choose it.

Reply:
Done.